plugins/woocommerce/src/Blocks/BlockTypes/ProductImage.php

```diff
@@ -102,13 +102,14 @@ private function render_on_sale_badge( $product, $attributes ) {
 			'
 		<div class="wc-block-components-product-sale-badge wc-block-components-product-sale-badge--align-%s wc-block-grid__product-onsale %s" style="%s">
 			<span aria-hidden="true">%s</span>
-			<span class="screen-reader-text">Product on sale</span>
+			<span class="screen-reader-text">%s</span>
 		</div>
 	',
 			esc_attr( $attributes['saleBadgeAlign'] ),
 			isset( $font_size['class'] ) ? esc_attr( $font_size['class'] ) : '',
 			isset( $font_size['style'] ) ? esc_attr( $font_size['style'] ) : '',
-			esc_html__( 'Sale', 'woocommerce' )
+			esc_html__( 'Sale', 'woocommerce' ),
+			esc_html__( 'Product on sale', 'woocommerce )
```

Comment:
We're missing a closing quotation here which will cause a parsing error across the site.

```suggestion
			esc_html__( 'Product on sale', 'woocommerce' )
```